### PR TITLE
Index is strftime formatted, fixes #454

### LINKED ISSUE #454: Add ability to handle index name template in ES output
So, as I understand correctly Fluentbit has an ability (if I use `Logstash_Format  True`) to get timestamp from incoming data (e.g. `@timestamp` field). But it would be great to set index template   time format (`index-name-%Y%m%d`) from local host not from incoming data. I don't want to use Logstash format, but I want to handle index name via time or env template from local host.

### PR DESCRIPTION
**Pull request by onorua**
fluentd does support index time templating, fluent bit - does not. This PR should fix that.

### CODE REVIEW

File: plugins/out_es/es.c [hunk 1]
```diff
@@ -189,10 +190,15 @@ static char *elasticsearch_format(void *data, size_t bytes,
...
+        flb_time_get(&tms);
+        gmtime_r(&tms.tm.tv_sec, &tm);
+        s = strftime(index_formatted, sizeof(index_formatted) - 1,
+                     ctx->index, &tm);
+        es_index = index_formatted;
         index_len = snprintf(j_index,
                              ES_BULK_HEADER,
                              ES_BULK_INDEX_FMT,
-                             ctx->index, ctx->type);
+                             es_index, ctx->type);
```
Comment (edsiper) [inline plugins/out_es/es.c]: is this ok when ctx->current_time_index == FLB_FALSE ?

**Comment by edsiper** [comment]
ping

**Comment by StevenACoffman** [comment]
@onorua I think you need to amend your commit with the `git -s` for the DCO check to pass.

Also, not sure why the tests are failing with this error:
```
[ 82%] Linking C executable ../bin/fluent-bit
../library/libflb-plugin-out_es.a(es_conf.c.o): In function `flb_es_conf_create':
/home/travis/build/fluent/fluent-bit/plugins/out_es/es_conf.c:253: undefined reference to `bool_value'
collect2: error: ld returned 1 exit status
make[2]: *** [bin/fluent-bit] Error 1
make[1]: *** [src/CMakeFiles/fluent-bit-bin.dir/all] Error 2
make: *** [all] Error 2
```
Your change seems not to touch this, and [the failing code block in your branch](https://github.com/onorua/fluent-bit/blob/454-strftime-for-index-name/plugins/out_es/es_conf.c#L260) isn't the same in master, so perhaps rebasing will fix it?

**Comment by edsiper** [comment]
@StevenACoffman have you tried the changes on this PR?

**Comment by edsiper** [comment]
@onorua in order to close/merge this one, would you please rebase your changes on top of GIT master ?

**Comment by onorua** [comment]
Sure, will do on Monday.

**Comment by onorua** [comment]
done

File: plugins/out_es/es.c (hunk 1 above)
Comment (StevenACoffman) [reply plugins/out_es/es.c]: @onorua Did you address this question?

Comment (StevenACoffman) [reply plugins/out_es/es.c]: Nevermind. Looks not relevant in the latest.

**Comment by edsiper** [comment]
PR merged as a separate commit with required changes: 455eae05

**Pull request closed without merging**